packages/backend/src/controllers/collection.controller.ts

```diff
@@ -0,0 +1,122 @@
+import { Request, Response } from 'express';
+import {
+  User,
+  UserOptionalDefaults,
+} from '@marvel-collector/types/generated/modelSchema/';
+import { string } from 'zod';
+import {
+  createCollectionItem,
+  existingComicInCollection,
+  findUniqueId,
+  queryCollectors,
+  viewCollections,
+} from '../services/collection.service';
+
+// Assigning Comics to a User collection
+
+export async function addCollectionItemToUser(req: Request, res: Response) {
+  const { id } = req.user as User;
+  const { comicId, title, imageUrl } = req.body;
+
+  try {
+    // Check if the user exists
+    const user = await findUniqueId(id);
+    if (!user) {
+      return res.status(404).json({ error: 'User not found' });
+    }
+
+    // Check if comicId already exists in User's collection
+
+    const existingCollectionItem = await existingComicInCollection(comicId, id);
+    if (existingCollectionItem) {
+      return res.status(400).json({
+        error: `Comic with id ${comicId} already exists in User's collection`,
+      });
+    }
+
+    // Create a new collection item
+    const collection = await createCollectionItem(id, comicId, title, imageUrl);
+
+    return res.status(200).json({
+      message: 'Collection item added to user collection successfully',
+      collection,
+    });
+  } catch (error) {
+    console.error(error);
+    return res.status(500).json({ error: 'Internal server error' });
+  }
+}
+
+// Endpoint for viewing comic book collections of a USER
+
+export async function viewComicBookCollectionOfUser(
+  req: Request,
+  res: Response,
+) {
+  const { id } = req.params as User;
+
+  try {
+    const user = await viewCollections(id);
+
+    // Check if user has any collections
+
+    if (!user?.collection.length) {
+      return res.status(404).json({ error: 'User has no collections' });
+    }
+    return res.status(200).json({
+      status: 'success',
+      data: {
+        id: user.id,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        username: user.username,
+        email: user.email,
+        profileImage: user.profileImage,
+        location: user.location,
+        collection: user.collection.map((item) => ({
+          id: item.id,
+          comicId: item.comicId,
+          title: item.title,
+          imageUrl: item.imageUrl,
+        })),
+      },
```

Comment:
Can we just return the user collection here. We dont need to return any user details.
If the collection is an empty array that's fine. 
```
res.json({
data: {
 collection: user.collection
}
})
```

Reply:
not sure if we need the "success" status? If the data is being returned then that suggests it's successful